File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/implicit/ImplicitAuthorizationRequest.java
package jp.co.sunarch.apps.idprovider.model.oauth2.implicit;

import jp.co.sunarch.apps.idprovider.model.client.ClientId;
import jp.co.sunarch.apps.idprovider.model.oauth2.RedirectUri;
import jp.co.sunarch.apps.idprovider.model.oauth2.ResponseType;
import jp.co.sunarch.apps.idprovider.model.oauth2.Scope;

import java.util.EnumSet;
import java.util.Set;

/**
 * @author takeshi
 */
public class ImplicitAuthorizationRequest {

  private ClientId clientId;
  private RedirectUri redirectUri;
  private Set<Scope> scope;
  private String state;

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/implicit/ImplicitAuthorizationRequest.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.implicit

import jp.co.sunarch.apps.idprovider.model.client.ClientId
import jp.co.sunarch.apps.idprovider.model.oauth2.RedirectUri
import jp.co.sunarch.apps.idprovider.model.oauth2.ResponseType
import jp.co.sunarch.apps.idprovider.model.oauth2.Scope

import java.util.EnumSet

/**
 * @author takeshi
 */
class ImplicitAuthorizationRequest {

  private val clientId: ClientId? = null
  private val redirectUri: RedirectUri? = null
  private val scope: Set<Scope>? = null
  private val state: String? = null

}
